src/Bicep.Core/Analyzers/Linter/Rules/NoUnusedRuleBase.cs

```diff
@@ -26,11 +26,24 @@ public abstract class NoUnusedRuleBase : LinterRuleBase
     protected AnalyzerFixableDiagnostic CreateRemoveUnusedDiagnosticForSpan(string name, IdentifierSyntax nameSyntax, SyntaxBase declaringSyntax, ImmutableArray<int> lineStarts, ProgramSyntax programSyntax)
     {
         var span = GetSpanForRow(programSyntax, declaringSyntax);
-        var codeFix = new CodeFix($"Remove unused {type}", true, CodeFixKind.QuickFix, new CodeReplacement(span, String.Empty));
+        var codeFix = new CodeFix(GetCodeFixDescription(), true, CodeFixKind.QuickFix, new CodeReplacement(span, String.Empty));
         var fixableDiagnosticForSpan = CreateFixableDiagnosticForSpan(nameSyntax.Span, codeFix, name);
         return fixableDiagnosticForSpan;
     }
 
+    private string GetCodeFixDescription()
```

Comment:
Ideally, "type" would be an enum, not a string.  Also this switch would then have an assert or exception on "default:".

I'll leave it up to you if you want to update.  Thanks!

Reply:
Good idea. I changed it to an enum.